lib/google/gax/api_callable.rb

```diff
@@ -372,10 +380,17 @@ def retryable(a_func, retry_options, kwargs)
     # @param a_func [Proc] a proc to be updated
     # @param timeout [Numeric] to be added to the original proc as it
     #   final positional arg.
+    # @param metadata [Hash] request metadata headers
     # @return [Proc] the original proc updated to the timeout arg
-    def add_timeout_arg(a_func, timeout, kwargs)
-      proc do |request|
-        a_func.call(request, deadline: Time.now + timeout, metadata: kwargs)
+    def add_timeout_arg(a_func, timeout, metadata)
+      proc do |request, block|
+        op = a_func.call(request,
+                         deadline: Time.now + timeout,
+                         metadata: metadata,
+                         return_op: true)
+        res = op.execute
+        block.call op if block
+        res
```

Comment:
A little bit confused that this works. If an `add_timeout` wraps a `retryable`, the `retryable` wrapping will cause the returned op to already call execute returning the result. If `add_timeout` does the same won't that throw an error?

Reply:
Not sure I'm following, but sounds like it could be an issue. Could you help with an example or test case to clarify?

Reply:
Lets say you have the method `foo` and you wrap it with retry and timeout like such:

`wrapped_foo = add_timeout( retryable(foo, ...) `

When `wrapped_foo` is called the call stack will look like:
```rb
foo_response = foo(request. return_op: true)
retryable_response = foo_response.execute
# this might error because execute has already been called to create the retryable response
add_timeout_timeout_response = retryable_response.execute
```

Reply:
Furthermore, if there was a method wrapped by both retryable and add_timeout, the passed block will be called twice once as it is processed by retryable and once when it is processed by add_timeout.

Reply:
Ah hm. I overlooked that the a method is either retryable or has timeout added so this might not be exactly a problem.